File: backend/AgriOpsAI.Api/Models/CropSeason.cs
namespace AgriOpsAI.Api.Models;

public enum CropSeasonStatus {planned, Active, Harvested, Closed}

public class CropSeason
{
    public Guid Id{get;set;}
    public Guid FieldId{get;set;}
    public Guid CropId{get;set;}
    public string SeasonName {get;set;} = string.Empty;
    public DateTime StartDate {get;set;} 
    public DateTime TargetEndDate {get;set;}
    public CropSeasonStatus Status {get;set;} = CropSeasonStatus.planned;

    public Field Field {get;set;} = null!;
    public Crop Crop {get;set;} = null!;
    public ICollection<Planting> Plantings {get;set;} = new List<Planting>();
    public ICollection<Harvest> Harvests {get;set;} = new List<Harvest>(); 
}

File: mobile/lib/models/crop_season.dart
class CropSeason {
  final String id;
  final String fieldId;
  final String cropId;
  final String seasonName;
  final DateTime startDate;
  final DateTime targetEndDate;
  final String status;
  final String? currentGrowthStage;

  CropSeason({
    required this.id,
    required this.fieldId,
    required this.cropId,
    required this.seasonName,
    required this.startDate,
    required this.targetEndDate,
    required this.status,
    this.currentGrowthStage,
  });

  factory CropSeason.fromJson(Map<String, dynamic> json) {
    return CropSeason(
      id: json['id'],
      fieldId: json['fieldId'],
      cropId: json['cropId'],
      seasonName: json['seasonName'],
      startDate: DateTime.parse(json['startDate']),
      targetEndDate: DateTime.parse(json['targetEndDate']),
      status: json['status'],
      currentGrowthStage: json['currentGrowthStage'],
    );
  }

  Map<String, dynamic> toCreateJson() {
    return {
      'fieldId': fieldId,
      'cropId': cropId,
      'seasonName': seasonName,
      'startDate': startDate.toIso8601String(),
      'targetEndDate': targetEndDate.toIso8601String(),
      'status': status,
    };
  }
}